blog/_posts/2016-10-06-julia-0.5-highlights.md

```diff
@@ -0,0 +1,726 @@
+Sometimes we can't do that so we just bite the bullet and break code with an error – so rude! But changing slices to views wouldn't break code with an error, it would just silently cause code to produce different, incorrect results.
+There's no clear way to make this transition safely.
+
+Taken together this makes a compelling case against changing the default slicing behavior to creating a view.
+That said, even if they're not the default, views are a crucial tool for performance in some situations.
+Accordingly, a huge amount of work went into improving the ergonomics of views in 0.5, including:
+
+- Renaming the function to create a view from "`sub`" to "`view`", which seems like a much better name.
+
+- Array views now support all forms of indexing supported by arrays, whereas previously there some more complex forms of indexing were not supported for views.
+
+- The `@view` macro was introduced, allowing the use of natural slicing syntax for array views.
+In other words you can write `@view A[1:end-1,2:end]` instead of `view(A, 1:size(A,1)-1, 2:size(A,2))`.
+
+Since views are an such important tool for both performance and for expressing complex mutating operations on arrays, especially with higher order functions, which are now fast, we may introduce a special syntax for view slices in the future.
+In particular, the syntax `A@[I...]` had a fair amount of popular support.
+Stay tuned!
+
+## And more...
+
+This is far from the full extent of the improvements introduced in Julia 0.5, but this blog post is already getting quite long, so I'll just summarize a few of the other big ticket items:
+
+- The set of string types and operations has been significantly simplified and streamlined.
+The `ASCIIString` and `UTF8String` types have been merged into a single `String` type, and the `UTF16String` and `UTF32String` and other types and functions related to string encodings other than UTF-8 have been moved into the `LegacyStrings` package, which keeps the same implementations as 0.4.
+In the future, better support for different string encodings will be developed under the `StringEncodings` package.
+
+- Most functionality related to prime generation, primality checking and combinatorics, has been moved into one of the two external packages: `Primes` or `Combinatorics`.
+To use these functions now you'll need to install the appropriate package and do `using Primes` or `using Combinatorics`.
+
+- Julia's LLVM version was upgraded from 3.3 to 3.7.1.
+This may not seem like a big deal, but the transition required herculean effort by many core Julia contributors.
+For a series of different and impossibly annoying reasons, LLVM versions 3.4 through 3.6 were not usable for Julia, so we're very happy to be back to using current versions of our favorite compiler framework.
+
+- Support for compiling and running Julia on [ARM] chips is much improved and 0.5 introduced initial support for [Power] systems, which we will be improving even further in the future.
```

Comment:
I've avoided mentioning any names in this post since once you start mentioning some people who did some work, you should mention everyone. So mentioning IBM's support feels a bit funny to me.